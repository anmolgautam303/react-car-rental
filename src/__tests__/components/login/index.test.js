import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Login from '../../../components/login';

configure({ adapter: new Adapter() })

describe('Joke', () => {
  const credentials = { email: 'anmol@test.com', password: 'pass' };
  let login = shallow(<Login />);

  it('renders properly', () => {
    expect(Login).toMatchSnapshot();
  });

  it('displays the email input', () => {
    expect(login.find('#input-auth-username').length).toBe(1);
  });

  it('displays the email input', () => {
    expect(login.find('#input-auth-password').length).toBe(1);
  });

  it('email field is filled correctly', () => {
    const emailInput = login.find('#input-auth-username');
    emailInput.value = credentials.email;
    expect(emailInput.value).toBe('anmol@test.com');
  });

  it('email field is filled correctly', () => {
    const passwordInput = login.find('#input-auth-password');
    passwordInput.value = credentials.password;
    expect(passwordInput.value).toBe('pass');
  });
});