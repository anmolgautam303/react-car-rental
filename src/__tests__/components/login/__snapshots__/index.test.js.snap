// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke renders properly 1`] = `[Function]`;
